cmd/maya-apiserver/app/server/volume_endpoint_v1alpha1.go

```diff
@@ -139,6 +139,12 @@ func (v *volumeAPIOpsV1alpha1) read(volumeName string) (*v1alpha1.CASVolume, err
 		vol.Namespace = hdrNS
 	}
 
+	// use sc name from header if present
+	scName := v.req.Header.Get(string(v1alpha1.StorageClassKey))
+	if scName != "" {
+		vol.Annotations[string(v1alpha1.StorageClassKey)] = scName
```

Comment:
This is not available. Remove these lines.
Remember a GET request i.e. read volume has name of volume & http request header

Reply:
This value needs to be set into the CASVolume object. Do a TrimSpace before setting.